### PR TITLE
rocksdb: support setting `ttl` and `periodic_compaction_seconds`.

### LINKED ISSUE #14873: rust-rocks: expose ColumnFamilyOptions's ttl and periodic_compaction_seconds setting in tikv
## Development Task
Recently, we found TTL compaction and periodical compaction can be too big to handle in time. 
We should expose the RocksDB's parameters in TiKV so that TiKV customers can opt to disable TTL and periodical compaction.

### PR DESCRIPTION
**Pull request by LykxSassinator**
### What is changed and how it works?

Issue Number: Close #14873 

What's Changed:

```commit-message
Expose the configuration settings on CF.ttl and CF.periodic_compaction_seconds.
```

### Related changes

- PR to update `pingcap/docs`/`pingcap/docs-cn`:
- Need to cherry-pick to the release branch

### Check List 

Tests 

- Unit test
- Integration test
- Manual test (add detailed scripts or steps below)
- No code

Side effects

- Performance regression
    - Consumes more CPU
    - Consumes more MEM
- Breaking backward compatibility

### Release note 

```release-note
None
```

### CODE REVIEW

File: src/config/mod.rs [hunk 1]
```diff
@@ -884,6 +900,8 @@ impl Default for WriteCfConfig {
             format_version: 2,
             checksum: ChecksumType::CRC32c,
             max_compactions: 0,
+            ttl: None,
```
Comment (tonyxuqqi) [inline src/config/mod.rs:903]: @tabokie Should we simply disable ttl and periodic compaction by default?

Comment (tabokie) [reply src/config/mod.rs:903]: I don't have strong opinion on this. The large compaction triggered by ttl will eventually be triggered by other reasons. The only difference is that ttl is more deterministic.

**Review by tabokie — COMMENTED** [review]
Need to update config template.

File: src/config/mod.rs (hunk 1 above)
Comment (tonyxuqqi) [reply src/config/mod.rs:903]: > I don't have strong opinion on this. The large compaction triggered by ttl will eventually be triggered by other reasons. The only difference is that ttl is more deterministic.

Is there any restriction when pick the input files for cross level compaction.  For example, by common sense it's not reasonable to compact a 4MB L5 file with 600GB L6 files, regardless of the reason.  I''m wondering if there's such restriction or can we add it?

**Review by tonyxuqqi — APPROVED** [review]

File: src/config/mod.rs (hunk 1 above)
Comment (tabokie) [reply src/config/mod.rs:903]: No. The restriction is dangerous. E.g. a SST containing range tombstone would overlap with lots of files, adding this restriction will make it impossible to free up space.

**Review by tabokie — APPROVED** [review]

**Pull request merged**